hawc/apps/assessment/models.py

```diff
@@ -775,11 +776,97 @@ def try_read_df(cls, data, suffix: str, worksheet_name: str = None) -> pd.DataFr
         return df
 
 
+class Job(models.Model):
+    PENDING = 1
+    SUCCESS = 2
+    FAILURE = 3
+    STATUS_CHOICES = (
+        (1, "PENDING"),
+        (2, "SUCCESS"),
+        (3, "FAILURE"),
+    )
+
+    TEST = 1
+    JOB_CHOICES = ((1, "TEST"),)
+    JOB_TO_FUNC = {
+        TEST: jobs.test,
+    }
+
+    task_id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+    assessment = models.ForeignKey(
+        Assessment, blank=True, null=True, on_delete=models.CASCADE, related_name="jobs"
+    )
+    status = models.PositiveSmallIntegerField(
+        choices=STATUS_CHOICES, default=PENDING, editable=False
+    )
+    job = models.PositiveSmallIntegerField(choices=JOB_CHOICES, default=TEST)
+
+    kwargs = JSONField(default=dict, blank=True, null=True)
+    result = JSONField(default=dict, editable=False)
+
+    created = models.DateTimeField(auto_now_add=True)
+    last_updated = models.DateTimeField(auto_now=True)
+
+    class Meta:
+        ordering = ("-created",)
+
+    def execute(self) -> Any:
+        """
+        Executes the job. Function and kwargs are determined
+        by the instance's "job" and "kwargs" properties, respectively.
+
+        Returns:
+            Any: Any data returned by the function call.
+            {"data" : <return_value>} MUST be JSON serializable.
+        """
+        func = self.get_func()
+        return func(**self.kwargs)
+
+    def get_func(self) -> Callable:
```

Comment:
do we need get_func? we could just add to execute, it's only used there, unless there's a reason to separate it